File: Hello_World_Tutorials/testing/c++/dllist.cpp
#include <iostream>
#include <string>

using namespace std;

#include "Bridges.h"
#include "DLelement.h"


using namespace bridges;

int main() {


	Bridges::initialize(4, "kalpathi60", "486749122386");
//	Bridges::initialize(4, "bridges_public", "997924677918");

	DLelement<string> *dl0 = new DLelement<string>("0", "5");
	DLelement<string> *dl1 = new DLelement<string>("1", "0");
	DLelement<string> *dl2 = new DLelement<string>("2", "1");
	DLelement<string> *dl3 = new DLelement<string>("3", "2");
        
	dl0->setNext(dl1);
	dl1->setNext(dl2);
	dl2->setNext(dl3);
	dl3->setPrev(dl2);
	dl2->setPrev(dl1);
	dl1->setPrev(dl0);


	dl0->getVisualizer()->setColor(Color("red"));
	dl1->getVisualizer()->setColor(Color("green"));
	dl2->getVisualizer()->setColor(Color("blue"));
	dl3->getVisualizer()->setColor(Color("yellow"));

	dl0->getLinkVisualizer(dl1)->setColor(Color("red"));
	dl1->getLinkVisualizer(dl2)->setColor(Color("green"));
	dl2->getLinkVisualizer(dl3)->setColor(Color("blue"));
	dl3->getLinkVisualizer(dl2)->setColor(Color("cyan"));
	dl2->getLinkVisualizer(dl1)->setColor(Color("magenta"));
	dl1->getLinkVisualizer(dl0)->setColor(Color("yellow"));

	Bridges::setTitle("Doubly Linked List Example");
	Bridges::setDataStructure(dl0);

	Bridges::visualize();

	return 0;
}

File: tutorials/testing/java/dllist.java
import bridges.base.DLelement;
import bridges.connect.Bridges;
import java.util.Iterator;

public class dllist {

	public static void main(String[] args) throws Exception {


		Bridges bridges = new Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID",
			"YOUR_API_KEY");

		// set title
		bridges.setTitle("A doubly Linked List Example");

		// set description
		bridges.setDescription("This list has five nodes all linked to the nodes before and after them and illustrates visual attributes. ");


		// create some elements
		DLelement<String>  el0 = new DLelement<String> ("Gretel Chaney", "Gretel Chaney");
		DLelement<String>  el1 = new DLelement<String> ("Lamont Kyler", "Lamont Kyler");
		DLelement<String>  el2 = new DLelement<String> ("Gladys Serino", "Gladys Serino");
		DLelement<String>  el3 = new DLelement<String> ("Karol Soderman", "Karol Soderman");
		DLelement<String>  el4 = new DLelement<String> ("Starr McGinn", "Starr McGinn");

		// create the list
		el0.setNext(el1);
		el1.setPrev(el0);
		el1.setNext(el2);
		el2.setPrev(el1);
		el2.setNext(el3);
		el3.setPrev(el2);
		el3.setNext(el4);
		el4.setPrev(el3);

		// test SLelement iterators

		// using a for-each loop
		System.out.println("Using range for loop:");
		for (String s : el0)
			System.out.println("\t " + s);

		// use the forward iterator
		System.out.println("Using forward iterators..");
		Iterator<String> iter = el0.iterator();
		while (iter.hasNext()) {
			System.out.println("\t" + iter.next());
		}

		// use the reverse iterator
		System.out.println("Using reverse iterator..");
		iter = el4.reverse_iterator();
		while (iter.hasNext()) {
			System.out.println("\t" + iter.next());
		}

		// add  element colors
		// set colors for list elements - see the Color class for supported colors
		el0.setColor("red");
		el2.setColor("aliceblue");

		// color the links - must specify a valid terminating element
		el0.getLinkVisualizer(el1).setColor("green");
		// color the reverse link
		el1.getLinkVisualizer(el0).setColor("magenta");

		// adjust link thickness
		el3.getLinkVisualizer(el4).setThickness(3.0f);
		el4.getLinkVisualizer(el3).setThickness(6.0f);

		// set link label
		el2.getLinkVisualizer(el3).setLabel("Link Label");

		// set node transparency
		el4.setOpacity (0.5f);

		// set node size
		el0.setSize (20);

		bridges.setDataStructure(el0);
		bridges.visualize();
	}
}
